### DISCUSSION TITLE
Help with error seting up Integration

### DISCUSSION THREAD

**Discussion by hoskerism** [nirnachmani/Pixie-Plus-for-Home-Assistant#8] [Q&A] [answered]
Hi All,

I'm having a strange problem installing the Integration in HA. I have installed the code in custom_components, but when I 'Add Integration' I get a login screen without any labels:
![image](https://github.com/nirnachmani/Pixie-Plus-for-Home-Assistant/assets/6742488/400bb23d-cb2e-4beb-bd17-d64abd2cec70)

I can work out which field is which by inspecting the html. But when I enter my credentials and submit I get an 'invalid_auth' exception:
![image](https://github.com/nirnachmani/Pixie-Plus-for-Home-Assistant/assets/6742488/ebf7252a-202f-413b-a8f0-8b2ad5c08717)

Note I tried using an online API tester and successfully connected using my credentials, so I don't think invalid credentials is the problem here.

Any help is appreciated!

**Comment by nirnachmani** [comment, marked as answer]
No idea why this is happening. I'll need to have a test HA and try install the integration to see if I can reproduce the problem but I really don't have time to do it currently. Maybe try the changes suggested by @brendanmckenzie (link - https://github.com/brendanmckenzie/Pixie-Plus-for-Home-Assistant/tree/main) - it may be related to the same problem he had were he needed to change one of the parameters (javascriptkey to clientkey) and they way the integration does HTTP requests.

**Reply by hoskerism** [reply]
Amazing - that was it!

It didn't fix the labels issue. (Although maybe that's a browser cache thing). But I can now connect and my devices are listed - which is awesome.

Many thanks.
